swarm/testutil/file.go

```diff
@@ -42,3 +44,18 @@ func TempFileWithContent(t *testing.T, content string) string {
 	}
 	return tempFile.Name()
 }
+
+// RandomBytes returns pseudo-random deterministic result
+// because test fails must be reproducible
+func RandomBytes(seed, length int) []byte {
+	source := mrand.NewSource(int64(seed))
+	b := make([]byte, length)
+	if _, err := mrand.New(source).Read(b); err != nil {
```

Comment:
i think this may not be safe for longer length since read is not guaranteed to read into the entire bufffer at once. 
you better just loop 

```golang
reader := rand.New(rand.NewSource(int64(seed)))
for n:= 0; n < length; { 
   if read, err := reader.Read(b[n:]); err  != nil {
     panic(err)
  }
  n+=read
}
```